app/main.py

```diff
@@ -1,8 +1,29 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person in people:
+        if "wife" in person and person["wife"]:
+            name = Person(person["name"], person["age"])
+            name.wife = person["wife"]
+
+        if "husband" in person and person["husband"]:
+            name = Person(person["name"], person["age"])
+            name.husband = person["husband"]
+        else:
+            name = Person(person["name"], person["age"])
+        person_list.append(name)
```

Comment:
you do redundant operations: double checks (Ex: `if "wife" in person and person["wife"]:`) and redundant assignments (Ex: `name.wife = person["wife"]` - you don't need it as `wife` should be of type `Person`, no need to assign something temporal).
Better write short first cycle where you just create instances of Person class and in the second cycle do what you do: check if wife/husband in person and assign her/him